nullaway/src/main/java/com/uber/nullaway/ClassAnnotationInfo.java

```diff
@@ -168,15 +169,22 @@ private CacheRecord get(Symbol.ClassSymbol classSymbol, Config config) {
   private boolean shouldTreatAsUnannotated(Symbol.ClassSymbol classSymbol, Config config) {
     if (config.isUnannotatedClass(classSymbol)) {
       return true;
-    } else if (config.shouldTreatGeneratedAsUnannotated()
-        && ASTHelpers.hasDirectAnnotationWithSimpleName(classSymbol, "Generated")) {
+    } else if (config.treatGeneratedAsUnannotated()) {
       // Generated code is or isn't excluded, depending on configuration
       // Note: In the future, we might want finer grain controls to distinguish code that is
-      // generated with nullability info and without.
-      return true;
-    } else {
-      return false;
+      // generated with nullability info and without. Possibly using the same mechanism as we
+      // will need to support @NullUnmarked.
+      if (ASTHelpers.hasDirectAnnotationWithSimpleName(classSymbol, "Generated")) {
+        return true;
+      }
+      ImmutableSet<String> generatedCodeAnnotations = config.getGeneratedCodeAnnotations();
```

Comment:
At first I was confused as to whether this would create a copy of `generatedCodeAnnotations` each time it is called.  But reading through the rest of the code, I think it won't, since the underlying `Set<String>` is actually an `ImmutableSet<String>`

Reply:
We do that for all our annotation sets in config. Might be worth replacing them all with `ImmutableSet<_>` and perf testing that!

Reply:
I think actually the `copyOf()` call is "free" since it is making a copy of another `ImmutableSet` (not evident from the types but that's what I think based on tracing through the code).  I presume it has a fast path for that case that just returns the original set